File: Data Structures & Algorithms/house-robber-ii/submission-0.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 1) {
            return nums[0];
        }
        else if (nums.length == 2) {
            int max = 0;
            for (int num : nums) {
                max = Math.max(max, num);
            }
            return max;
        }
        int[] arr = new int[nums.length - 1];
        arr[0] = nums[0];
        arr[1] = Math.max(arr[0], nums[1]);
        for (int i = 2; i < nums.length - 1; i++) {
            arr[i] = Math.max(arr[i-2] + nums[i], arr[i-1]);
        }
        int first = arr[nums.length - 2];
        arr[0] = nums[1];
        arr[1] = Math.max(nums[1], nums[2]);
        for (int i = 2; i < nums.length-1; i++) {
            arr[i] = Math.max(arr[i-2] + nums[i+1], arr[i-1]);
        }
        return Math.max(first, arr[nums.length - 2]);

    }
}

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-0.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        arr = [0] * 3 
        for i in range(len(triplets)):
            first = triplets[i][0]
            second = triplets[i][1]
            third = triplets[i][2]
            if first <= target[0] and second <= target[1] and third <= target[2]:
                arr[0] = max(first, arr[0])
                arr[1] = max(second, arr[1])
                arr[2] = max(third, arr[2])
        for i in range(len(arr)):
            if target[i] != arr[i]:
                return False
        return True
        